runtime/src/main/java/org/corfudb/runtime/collections/Table.java

```diff
@@ -60,12 +61,24 @@ public class Table<K extends Message, V extends Message, M extends Message> impl
     // gets block on parallel stream, because the pool is exhausted with threads that are trying to acquire the VLO
     // look, which creates a circular dependency. In other words, a deadlock.
 
-    protected static final ForkJoinPool pool = new ForkJoinPool(Math.max(Runtime.getRuntime().availableProcessors() / 2, 1),
+    private static final int FJP_PARALLELISM = Math.max(Runtime.getRuntime().availableProcessors() / 2, 1);
+
+    protected static final ForkJoinPool pool = new ForkJoinPool(
+            FJP_PARALLELISM,
             pool -> {
                 final ForkJoinWorkerThread worker = ForkJoinPool.defaultForkJoinWorkerThreadFactory.newThread(pool);
                 worker.setName("Table-Forkjoin-pool-" + worker.getPoolIndex());
                 return worker;
-            }, null, true);
+            },
+            null,
+            true,
+            2 * FJP_PARALLELISM, // corePoolSize == 2 * parallelism to avoid bug JDK-8330017
+            Integer.MAX_VALUE, // default to FJP.MAX_CAP
+            1,
+            null,
```

Comment:
Would it be possible to add a comment saying why we need these parameters?

Reply:
All the additional parameters have FJP constructor's default values. Note that this is the only constructor that allows you to set the corePoolSize so it looks verbose.